### ISSUE TITLE
git submodule USERAUTH fail when private repositories

### ISSUE THREAD

**Issue by huguesfontenelle** [nextflow-io/nextflow#1028]
## Bug report 

My repository is private.
It has a *submodule* which is also private.
I specify the credentials in SCM configuration file.
`nextflow pull repo/pipeline`
pulls the pipeline but not the submodule.

### Expected behavior and actual behavior

(Submodules should be authenticated)

### Steps to reproduce the problem

(By definition, it is a bit difficult to share private repositories)

### Program output 

```
$ nextflow pull repo/pipeline 
Checking repo/pipeline ...
git@github.com:repo/data.git: USERAUTH fail
```
Where `repo/data` is included as a submodule in `repo/pipeline`.

### Environment 

* Nextflow version: 19.01.0
* Java version: 11.0.2
* Operating system: macOS

### Additional context

Contents of `.gitmodules`:
```
[submodule "data"]
	path = data
	url = git@github.com:repo/data.git
	branch = dev
```

Note: changing the authentication from git to https, ie changing the url to `https://github.com/repo/data.git` **and** making `data` pubic works. Obviously.

**Comment by rspreafico-vir** [comment]
Confirming this with `19.0.0.modules-draft10-SNAPSHOT`, with the caveat that in my case it does not attempt to download the submodules. (for me too, both parent and child repo are private)

This will become a more important issues when Nextflow modules are finalized, since a natural way to distribute modules will be through `git submodule`.

**Comment by czotti** [comment]
Same here I have a pipeline with a `bin` folder which is a `git submodule` and I face an error like:
```
UnknownHostKey: hostname. RSA key fingerprint is XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX:XX
```

Could be really nice to have.

**Comment by huguesfontenelle** [comment]
I think this is solved in https://github.com/nextflow-io/nextflow/pull/1205, released in v19.07.0
